models/Shift.js

```diff
@@ -17,12 +18,27 @@ const ShiftSchema= new mongoose.Schema({
   },
   wage: Number,
   tips: Boolean,
-  date: Date,
-  start_time: String,
-  end_time: String,
-  activeStatus: { type: Boolean, default: true }, 
+  start_at: Date,
+  end_at: Date,
+  activeStatus: { type: Boolean, default: true },
   more: String,
   availability: [ { type: String } ]
 });
 
+ShiftSchema.virtual('duration').
+ get(function() {
+  const shiftLength = date.subtract(new Date(this.end_at), new Date(this.start_at)).toHours()
+  return Math.round( shiftLength * 100 ) / 100
+})
+
+ShiftSchema.virtual('start_time').
+ get(function() {

```

Comment:
You don't need these properties. Formatting of the dates is more a question for the templates. You should try to pass around regular `Date` objects as much as possible, because they are flexible and more useful than strings. Generally I would only format dates to a string in a template, or if I was writing an api and needed to send the date in a JSON payload.

Reply:
Sorry, I don't understand what you mean by template here. Could you give me an example of what's considered a template? 
In the case of this app, I want my time to be displayed in `hh:mm` format. Where do I process the formatting of my shift Date object into a string? I could only think of doing it in my `controller/accounts.js` in `getDashboard` method.